app/packages/core/src/components/ColorModal/JSONViewer.tsx

```diff
@@ -78,10 +79,24 @@ const JSONViewer: React.FC = () => {
         ? data?.showSkeletons
         : colorScheme?.showSkeletons
     );
+    const validatedLabelTags = {
+      fieldColor: isValidColor(data?.labelTags?.fieldColor)
+        ? colorString.to.hex(
+            colorString.get(data?.labelTags?.fieldColor as string)!.value
+          )
+        : undefined,
+      valueColors: data?.labelTags?.valueColors
+        ?.filter((pair) => isValidColor(pair.color))
+        .map((pair) => ({
+          color: colorString.to.hex(colorString.get(pair.color)!.value),
+          value: pair.value,
+        })),
```

Comment:
nit: create a utility function using useCallback outside of this method and have that function return hex value or undefined